### DISCUSSION TITLE
How to access the "Eyedropper" in colorpicker?

### DISCUSSION THREAD

**Discussion by WileyNg** [picoe/Eto#2212] [General]
Hi @cwensley,
many thanks for helping me with the various questions.
I have another problem...
I am trying to re-create a custom ColorPicker with a better dropdown.
I have got everything except I couldn't figure out how to call the Eyedropper like you did with Eto ColorPicker.

![image](https://user-images.githubusercontent.com/5993201/166688968-033bc1b1-3791-4d26-8aff-b08207631bea.png)

If I need to call an Eyedropper after a MouseUp event...  should I call it directly from Windows Form?  if yes, what about on Mac?

Any pointers would be very much appreciated!

**Comment by cwensley** [comment]
Hey @WileyNg, just for future reference, any questions relating to Rhino should go to discourse.mcneel.com, you'll get better support there.

On Windows, that uses the `RhinoWindows.Controls.GetColorFromScreen` class.  There is no public direct equivalent on Mac though.

As for the standard Rhino color dialog, it does it purely with Eto by constructing a Bitmap with contents from the `Screen.GetImage()` API, creating a Cursor, then using `Mouse.SetCursor()` to set the cursor icon to create the little magnification bubble.  You should be able to use the same APIs to implement your own variation of an eye dropper.
